pkg/deploy/controller/deploymentconfig/factory.go

```diff
@@ -0,0 +1,67 @@
+package deploymentconfig
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	kutil "github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	controller "github.com/openshift/origin/pkg/controller"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+// DeploymentConfigControllerFactory can create a DeploymentConfigController which obtains
+// DeploymentConfigs from a queue populated from a watch of all DeploymentConfigs.
+type DeploymentConfigControllerFactory struct {
+	Client     *osclient.Client
+	KubeClient kclient.Interface
+	Codec      runtime.Codec
+}
+
+func (factory *DeploymentConfigControllerFactory) Create() controller.RunnableController {
+	deploymentConfigLW := &deployutil.ListWatcherImpl{
+		ListFunc: func() (runtime.Object, error) {
+			return factory.Client.DeploymentConfigs(kapi.NamespaceAll).List(labels.Everything(), labels.Everything())
+		},
+		WatchFunc: func(resourceVersion string) (watch.Interface, error) {
+			return factory.Client.DeploymentConfigs(kapi.NamespaceAll).Watch(labels.Everything(), labels.Everything(), resourceVersion)
+		},
+	}
+	queue := cache.NewFIFO(cache.MetaNamespaceKeyFunc)
+	cache.NewReflector(deploymentConfigLW, &deployapi.DeploymentConfig{}, queue).Run()
+
+	configController := &DeploymentConfigController{
+		deploymentClient: &deploymentClientImpl{
+			getDeploymentFunc: func(namespace, name string) (*kapi.ReplicationController, error) {
+				return factory.KubeClient.ReplicationControllers(namespace).Get(name)
+			},
+			createDeploymentFunc: func(namespace string, deployment *kapi.ReplicationController) (*kapi.ReplicationController, error) {
+				return factory.KubeClient.ReplicationControllers(namespace).Create(deployment)
+			},
+		},
+		makeDeployment: func(config *deployapi.DeploymentConfig) (*kapi.ReplicationController, error) {
+			return deployutil.MakeDeployment(config, factory.Codec)
+		},
+	}
+
+	return &controller.RetryController{
+		Queue:        queue,
+		RetryManager: controller.NewQueueRetryManager(queue, cache.MetaNamespaceKeyFunc, 1),
+		ShouldRetry: func(obj interface{}, err error) bool {
+			if _, isFatal := err.(fatalError); isFatal {
+				return false
+			}
+			kutil.HandleError(err)
```

Comment:
seems backwards that you're logging the error when you retry, but not logging it if it's fatal.